truss/templates/control/control/server.py

```diff
@@ -73,4 +80,21 @@ def run(self):
         control_server_port=CONTROL_SERVER_PORT,
         inference_server_port=INFERENCE_SERVER_PORT,
     )
-    control_server.run()
+    InferenceServerReverseProxyPlugin.inference_server_path = (
+        f"http://localhost:{INFERENCE_SERVER_PORT}/".encode()
+    )
+    InferenceServerReverseProxyPlugin.control_server_path = (
+        f"http://localhost:{CONTROL_SERVER_PORT}/".encode()
+    )
+    InferenceServerReverseProxyPlugin.server_down_callback = (
+        is_inference_server_intentionally_stopped
+    )
+
+    with Proxy(
+        reversed=True,
+        threadless=True,
+        port=PROXY_SERVER_PORT,
+        plugins=[InferenceServerReverseProxyPlugin],
+        input_args=["--enable-reverse-proxy"],
+    ):
+        control_server.run()
```

Comment:
i know this is very much wip right now, but I think that eventually we should not use this decorator, and instead run it manually. It's not very obvious from this code that there are two servers running here.